instructor/ui/instructor.js

```diff
@@ -631,48 +663,107 @@ $(function() {
   });
 
   /* Open Image Dialog in Microscopy Analyze page */
+
   $(".open_upload_window_btn").click(function(){
     /* this btn has the id of the corresponding row */
     var row_id = $(this).data('row_id');
-    var filter_group_id = $(this).data('filter_group_id');
     /* Get name of the sample from the row itself */
     var sample_name = $("#" + row_id).text().replace(/(\n *)+/g, "");
     $('.scb_ab_f_sample_name').text(sample_name);
     var instance_pk = row_id.match(/(\d+)$/)[0];
-    var sample_treatment  = $(this).data('sample_treatment');
+    var sample_treatment  = $(this).data('sample_treatment'),
+        group_id = $(this).data('group_id') ? $(this).data('group_id'): null,
+        analysis = sample_treatment.split(',')[0];
+
     $('.scb_ab_f_treatment_text').text(sample_treatment);
     $('.scb_ab_s_analyze_dialog').css('visibility', 'visible');
     $('.scb_ab_f_save_image').data({
-      'pk': instance_pk,
-      'filter_group_id': filter_group_id
+        'pk': instance_pk,
+        'group_id': group_id,
+        'analysis': analysis,
     });
-    addImageFormHandler();
+    addSelectedImages(); // Add already saved images to the selected area
+  });
 
+  $(function(){
+   addSelectedImages(); // Add already saved images to the selected area on the page loading
   });
+
+  /* Remove remove selection from the images and revert all unsaved images to the image bank */
+  function clearSelectedFrame() {
+      var reselectImages = $(".scb_ab_s_small_image_selected");
+      reselectImages.each(function () {
+          $(this).removeClass("scb_ab_s_small_image_selected");
+          $(".scb_ab_s_image_bank").append($(this));
+      });
+  }
+
+  /* Fulfil filter area by chosen image in the Fluorescent image sececting flow*/
```

Comment:
Fulfil => Fulfill and  `sececting` => `selecting`

Reply:
fixed